ui/app/components/modal/modal.component.js

```diff
@@ -0,0 +1,94 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import Button from '../button'
+
+export default class Modal extends PureComponent {
+  static propTypes = {
+    children: PropTypes.node,
+    // Header text
+    headerText: PropTypes.string,
+    // Submit button (right button)
+    onSubmit: PropTypes.func,
+    submitType: PropTypes.string,
+    submitText: PropTypes.string,
+    // Cancel button (left button)
+    onCancel: PropTypes.func,
+    cancelType: PropTypes.string,
+    cancelText: PropTypes.string,
+  }
+
+  static defaultProps = {
+    submitType: 'primary',
+    cancelType: 'default',
+  }
+
+  handleClose = () => {
+    const { onCancel, onSubmit } = this.props
+
+    /**
+     * The close button should be used to dismiss the modal, without performing any actions, which
+     * is typically what props.onCancel does. However, if props.onCancel is undefined, that should
+     * mean that the modal is a simple notification modal and props.onSubmit can be used to dismiss
+     * it.
+     */
+    if (onCancel && typeof onCancel === 'function') {
```

Comment:
Yeah I was trying to simplify the API by not introducing an additional `onClose` prop since I thought in every use case the `onClose` would be the `onCancel` function if present (in modals where both a Cancel and a Submit button are shown) or be the `onSubmit` function (in modals where there's a single button in the footer). I can see why that can be confusing though.

Reply:
I'd also for some sort of change to this default functionality. I think that `onCancel` and `onSubmit` should be very distinct and not accidentally interchangable. I would prefer the user of the modal component have to submit the same function for onCancel and onSubmit if they want both to do the same.

One idea: what if the close button just doesn't appear when `onCancel` is undefined?

Reply:
I guess that is what you do with the "Cancel" button itself...

So, I'd vote for either adding a 3rd `onClose` prop, OR having the 'x' and the "Cancel" button always do the same thing, and be shown / hidden under the same conditions